x-pack/plugins/observability_solution/slo/public/embeddable/slo/error_budget/error_budget_react_embeddable_factory.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { i18n } from '@kbn/i18n';
+import React, { useEffect } from 'react';
+import { Router } from '@kbn/shared-ux-router';
+import { createBrowserHistory } from 'history';
+import { ReactEmbeddableFactory } from '@kbn/embeddable-plugin/public';
+import {
+  initializeTitles,
+  useStateFromPublishingSubject,
+  fetch$,
+} from '@kbn/presentation-publishing';
+import { BehaviorSubject, Subject } from 'rxjs';
+import { KibanaContextProvider } from '@kbn/kibana-react-plugin/public';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { SLO_ERROR_BUDGET_ID } from './constants';
+import { SloErrorBudgetEmbeddableState, SloEmbeddableDeps, ErrorBudgetApi } from './types';
+import { SloErrorBudget } from './error_budget_burn_down';
+
+export const getErrorBudgetPanelTitle = () =>
+  i18n.translate('xpack.slo.errorBudgetEmbeddable.title', {
+    defaultMessage: 'SLO Error Budget burn down',
+  });
+
+export const getErrorBudgetEmbeddableFactory = (deps: SloEmbeddableDeps) => {
+  const factory: ReactEmbeddableFactory<SloErrorBudgetEmbeddableState, ErrorBudgetApi> = {
+    type: SLO_ERROR_BUDGET_ID,
+    deserializeState: (state) => {
+      return state.rawState as SloErrorBudgetEmbeddableState;
+    },
+    buildEmbeddable: async (state, buildApi, uuid, parentApi) => {
+      // console.log(parentApi.renderComplete.dispatchComplete, '!!parentApi');
+      const { titlesApi, titleComparators, serializeTitles } = initializeTitles(state);
+      const defaultTitle = getErrorBudgetPanelTitle();
+      titlesApi.setPanelTitle(state.title ?? defaultTitle);
+      const sloId$ = new BehaviorSubject(state.sloId);
+      const sloInstanceId$ = new BehaviorSubject(state.sloInstanceId);
+      const reload$ = new Subject<boolean>();
+
+      const api = buildApi(
+        {
+          ...titlesApi,
+          serializeState: () => {
+            return {
+              rawState: {
+                ...serializeTitles(),
+                sloId: sloId$.getValue(),
+                sloInstanceId: sloInstanceId$.getValue(),
+              },
+            };
+          },
+        },
+        {
+          sloId: [sloId$, (value) => sloId$.next(value)],
+          sloInstanceId: [sloInstanceId$, (value) => sloInstanceId$.next(value)],
+          ...titleComparators,
+        }
+      );
+
+      const fetchSubscription = fetch$(api)
+        .pipe()
+        .subscribe((next) => {
+          reload$.next(next.isReload);
```

Comment:
This handles the manual refresh, and all works fine!

I have a question though. Above in the `packages/presentation/presentation_publishing/index.ts` I exported `export { apiPublishesReload, type PublishesReload } from './interfaces/fetch/publishes_reload';`, so that I could reuse it in the factory. I didn't know how. Is there a best practice regarding how to handle manual refresh? Is it fine how I did it?

Reply:
Yes, it looks like this is the correct way to do what you need! Is it expected that your component only refreshes on a manual refresh? I suppose so if it doesn't listen to any of the Dashboard state.

Reply:
Indeed our component only refreshes on a manual refresh. It is expected for now. After we migrate our embeddables to the new Architecture, our focus will be to make them more dynamic and react to [date picker](https://github.com/elastic/kibana/issues/181498), [controls](https://github.com/elastic/kibana/issues/181426) etc.